Windows agent failed to restart in AR tests

## Description

During testing in[ Release 4.8.0 - Alpha 1 IT](https://github.com/wazuh/wazuh/issues/20686) tests, the Windows AR test `test_execd_restart.py` failed.

```
set_debug_mode = None
get_configuration = {'apply_to_modules': ['test_execd_restart', 'test_execd_firewall_drop', 'conftest'], 'metadata': {'command': 'restart-...], 'section': 'client'}, {'elements': [{'disabled': {'value': 'no'}}], 'section': 'active-response'}], 'tags': ['all']}
test_version = None, configure_environment = None, start_agent = True
set_ar_conf_mode = None

    def test_execd_restart(set_debug_mode, get_configuration, test_version,
                           configure_environment, start_agent, set_ar_conf_mode):
        '''
        description: Check if 'restart-wazuh' command of 'active response' is executed correctly.
                     For this purpose, a simulated agent is used, to which the active response is sent.
                     This response includes the order to restart the Wazuh agent,
                     which must restart after receiving this response.
    
        wazuh_min_version: 4.2.0
    
        tier: 0
    
        parameters:
            - set_debug_mode:
                type: fixture
                brief: Set execd daemon in debug mode.
            - get_configuration:
                type: fixture
                brief: Get configurations from the module.
            - test_version:
                type: fixture
                brief: Validate Wazuh version.
            - configure_environment:
                type: fixture
                brief: Configure a custom environment for testing.
            - start_agent:
                type: fixture
                brief: Create Remoted and Authd simulators, register agent and start it.
            - set_ar_conf_mode:
                type: fixture
                brief: Configure Active Responses used in tests.
    
        assertions:
            - Verify that the 'active response' 'restart-wazuh' is received.
            - Verify that the agent is ready to restart.
    
        input_description: Different use cases are found in the test module and include
                           parameters for 'restart-wazuh' command and the expected result.
    
        expected_output:
            - r'DEBUG: Received message'
            - r'Shutdown received. Deleting responses.'
            - r'Starting'
            - r'active-response/bin/restart-wazuh'
            - r'Ended'
            - r'Invalid input format' (If the 'active response' fails)
    
        tags:
            - simulator
        '''
    
        # Check if the agent is restarted properly"
>       assert not start_agent, 'The agent failed to restart successfully after enrolling the authentication simulator.'
E       AssertionError: The agent failed to restart successfully after enrolling the authentication simulator.
E       assert not True

test_active_response\test_execd\test_execd_restart.py:246: AssertionError
 -----------------------------Captured stdout setup------------------------------ 
The Wazuh service was stopped successfully.

The Wazuh service is starting.

Failure to restart the agent

 -----------------------------Captured stderr setup------------------------------ 
2023/12/07 18:17:29 agent-auth: INFO: Started (pid: 3732).
2023/12/07 18:17:29 agent-auth: INFO: Requesting a key from server: 127.0.0.1
2023/12/07 18:17:30 agent-auth: INFO: No authentication password provided
2023/12/07 18:17:30 agent-auth: INFO: Using agent name as: EC2AMAZ-NRK2S46
```

- **Build:** https://ci.wazuh.info/job/Test_integration/45295/
- [Build artifact](https://s3.amazonaws.com/ci.wazuh.com/data/Test_integration/B45295/agent_windows_html_report_Test_integration_B45295.zip)


### Current behavior

The test fails to detect the agent restarting.

--- a/deps/wazuh_testing/wazuh_testing/tools/services.py
+++ b/deps/wazuh_testing/wazuh_testing/tools/services.py
@@ -88,7 +88,9 @@ def control_service(action, daemon=None, debug_mode=False):
             for _ in range(error_109_windows_retry):
                 command = subprocess.run(["net", action, "WazuhSvc"], stderr=subprocess.PIPE)
                 result = command.returncode
-                if result != 0:
+                if result == 0:
+                    break
+                else:
                     if action == 'stop' and 'The Wazuh service is not started.' in command.stderr.decode():
                         result = 0
                         break

